pkg/unidling/api/types.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import (
+	kextapi "k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
+	IdledAtAnnotation      = "idling.alpha.openshift.io/idled-at"
+	UnidleTargetAnnotation = "idling.alpha.openshift.io/unidle-targets"
+	NeedPodsReason         = "NeedPods"
+)
+
+// RecordedScaleReference is a SubresourceReference to a scale subresource that also
+// has the previous replica count recorded
+type RecordedScaleReference struct {
+	// TODO: update this to a CrossVersionObjectReference once we pull in a Kube version
+	// with an unversioned apis/autoscaling
+	kextapi.SubresourceReference `json:",inline"`
```

Comment:
Feel free to correct me if I'm wrong here, but that suggestion seems to completely go against existing API design -- all the other reference types, AFAICT, refer to `Kind` instead of `Resource` and `APIVersion` in the slash-separated form (e.g. ObjectReference, OwnerReference, and group-versions are written as `APIVersion` in `TypeMeta` as well).  Furthermore, the scale client interfaces refer to `Kind` as well, so if we took a resource, we'd just have to convert it back to kind anyway (furthermore, I don't think we want to explicitly state that we're referring to the scale subresource here -- just an object that happens to have a scale subresource.  I'd much rather use `unversioned.CrossVersionObjectReference`, but that wasn't in the rebase I was working on).

While I'm not necessarily against the changes you're suggesting here, it seems to me like it would be confusing at best to have yet another object reference type (especially since CrossVersionObjectReference and SubresourceReference were specifically designed for referring to things that needed to be scaled) that seems like it would be completely different than all the other reference types, and would use conventions different than the rest of the API.